### ISSUE TITLE
"Area code not set" when running console command

### ISSUE THREAD

**Issue by jsakars** [magento/magento2#5614]
## Steps to reproduce
1. Install Magento 2.1
2. Have custom module which has console command to import products from 3rd party providers
3. Run console command
## Expected result
1. Products are stored
## Actual result
1. Error occurs - `Area code not set: Area code must be set before starting a session.`

If I set `$state->setAreaCode('adminhtml');` - product import goes fine but then `setup:di:compile` gives `Area code is already set` error.
Version 2.0.7 had the same issue with the import command but compile command still worked fine.

If I comment out `$product->save();` then such error is not present.

**Comment by andimov** [comment]
@werdlv 
Can't reproduce this bug.
Please, add additional info for reproducing or check custom module.

**Comment by jsakars** [comment]
@andimov not sure what other additional info can I add

**Comment by slopukhov** [comment]
@werdlv thank you for your feedback.
The error may be at the custom module realization, please add describe or example where it sets $state->setAreaCode

**Comment by vzabaznov** [comment]
Hi @werdlv , closed as we don't receive update more than 2 week, feel free to write here/reopen or to open new issue
